crates/hir-ty/src/infer.rs

```diff
@@ -575,6 +579,8 @@ impl<'a> InferenceContext<'a> {
     // used this function for another workaround, mention it here. If you really need this function and believe that
     // there is no problem in it being `pub(crate)`, remove this comment.
     pub(crate) fn resolve_all(self) -> InferenceResult {
+        // NOTE: `InferenceResult::closure_info` is `resolve_completely()`'d during
+        // `InferenceContext::infer_closures()` (in `HirPlace::ty()` specifically).
         let InferenceContext { mut table, mut result, .. } = self;
```

Comment:
it might help to destructure `result` here so that when a new field is added it will error here

Reply:
Great idea, I'll do that!